src/hexer/nifcgen.nim

```diff
@@ -1184,13 +1184,33 @@ proc traverseStmt(e: var EContext; c: var Cursor; mode = TraverseAll) =
   else:
     error e, "statement expected, but got: ", c
 
+proc transformInlineRoutines(e: var EContext; c: var Cursor) =
+  var swapped = createTokenBuf()
+  swap e.dest, swapped
+
+  var toTransform = createTokenBuf()
+  toTransform.copyIntoKind StmtsS, c.info:
+    takeTree(toTransform, c)
+  var c0 = beginRead(toTransform)
+  var dest = transform(e, c0, e.main)
+  c = beginRead(dest)
```

Comment:
This is not correct, the underlying buffer does not live enough. I think you merely got lucky that the memory is not reused too soon.